cmd/prow-job-dispatcher/main.go

```diff
@@ -661,38 +672,51 @@ func cleanup(directory string) {
 	logrus.WithField("directory", directory).Info("Successfully removed directory")
 }
 
+// createPR creates PR with config changes and sanitizer changes, it causes app to exit in
+// case of failure to trigger re-run of logic
 func createPR(o options, config *dispatcher.Config, pjs map[string]string) {
 	targetDirWithRelease := filepath.Join(o.targetDir, "/release")
 	cleanup(targetDirWithRelease)
 	defer cleanup(targetDirWithRelease)
 
 	logrus.WithField("targetDir", o.targetDir).Info("Changing working directory ...")
 	if err := os.Chdir(o.targetDir); err != nil {
-		logrus.WithError(err).Error("failed to change to root dir")
-		return
+		logrus.WithError(err).Fatal("failed to change to root dir")
 	}
 
 	if err := gitCloneRelease(); err != nil {
-		logrus.WithError(err).Error("failed to clone release repository")
-		return
+		logrus.WithError(err).Fatal("failed to clone release repository")
 	}
 
 	if err := dispatcher.SaveConfig(config, filepath.Join(targetDirWithRelease, "/core-services/sanitize-prow-jobs/_config.yaml")); err != nil {
-		logrus.WithError(err).Errorf("failed to save config file to %s", o.configPath)
-		return
+		logrus.WithError(err).WithField("configPath", o.configPath).Fatal("failed to save config file")
 	}
 
 	if err := sanitizer.DeterminizeJobs(filepath.Join(targetDirWithRelease, "/ci-operator/jobs"), config, pjs); err != nil {
-		logrus.WithError(err).Error("failed to determinize")
-		return
+		logrus.WithError(err).Fatal("failed to determinize")
 	}
 
 	title := fmt.Sprintf("%s at %s", matchTitle, time.Now().Format(time.RFC1123))
 	if err := o.PRCreationOptions.UpsertPR(targetDirWithRelease, githubOrg, githubRepo, upstreamBranch, title, prcreation.PrAssignee(o.assign), prcreation.MatchTitle(matchTitle), prcreation.AdditionalLabels([]string{rehearse.RehearsalsAckLabel})); err != nil {
-		logrus.WithError(err).Error("failed to upsert PR")
+		logrus.WithError(err).Fatal("failed to upsert PR")
 	}
 }
 
+func sendSlackMessage(slackClient slackClient, channelId string) error {
+	blockMessage := slack.MsgOptionBlocks(
+		slack.NewSectionBlock(
+			slack.NewTextBlockObject("mrkdwn", fmt.Sprintf("*Scheduling PR to merge*\n\n<@dptp-triage> Prow jobs have been rescheduled. To ensure the proper functioning of legacy tooling, please prioritize merging PRs from this *<%s|list>*.", listURL), false, false),
```

Comment:
A direct ping, how frequently does that happen?

Reply:
every reshuffle so:
1. during outage when we change config
2. periodically on Monday morning

Reply:
> during outage when we change config

that's ok

> periodically on Monday morning

it makes me feel anxious

Reply:
I guess I will have to just live with it.

Reply:
PR must be merged regardless this is outage or periodic thing.